lib/src/js/compile.dart

```diff
@@ -182,6 +185,23 @@ OutputStyle _parseOutputStyle(String? style) => switch (style) {
 /// Converts [importer] into an [AsyncImporter] that can be used with
 /// [compileAsync] or [compileStringAsync].
 AsyncImporter _parseAsyncImporter(Object? importer) {
+  if (importer is NodePackageImporterClass) {
+    if (isBrowser) {
+      jsThrow(JsError(
+          "The Node Package Importer cannot be used without a filesystem."));
+    }
+    var entryPointPath = importer.entryPointPath != null
+        ? p.join(p.current, importer.entryPointPath)
+        : requireMainFilename;
+
+    if (entryPointPath == null) {
+      jsThrow(JsError(
+          "The Node Package Importer cannot determine an entry point."
+          "Please provide an `entryPointPath` to the Node Package Importer."));
```

Comment:
This message should indicate that this is because `require.main.filename` isn't defined.

Reply:
Addressed in https://github.com/oddbird/dart-sass/commit/15e33b9b2fc0a9848076f2356a1c99680f8198bc